app/dao/inbound_sms_dao.py

```diff
@@ -62,6 +63,34 @@ def dao_count_inbound_sms_for_service(service_id, limit_days):
     ).count()
 
 
+def _insert_update_inbound_sms_history(subquery, query_limit=10000):
+    offset = 0
+    inbound_sms_query = db.session.query(
+        *[x.name for x in InboundSmsHistory.__table__.c]
+    ).filter(InboundSms.id.in_(subquery))
+    inbound_sms_count = inbound_sms_query.count()
+
+    while offset < inbound_sms_count:
+        statement = insert(InboundSmsHistory).from_select(
+            InboundSmsHistory.__table__.c,
+            inbound_sms_query.limit(query_limit).offset(offset)
+        )
+
+        statement = statement.on_conflict_do_update(
+            constraint="inbound_sms_history_pkey",
+            set_={
+                "service_id": statement.excluded.service_id,
+                "notify_number": statement.excluded.notify_number,
+                "provider_date": statement.excluded.provider_date,
+                "provider_reference": statement.excluded.provider_reference,
+                "provider": statement.excluded.provider
+            }
+        )
```

Comment:
@leohemsted 

I'm wondering whether these lines are adding value. They aren't being tested (I've removed them from our code and no tests fail).

Further, I'm trying to work out in which scenario we expect them to run. A conflict seems unlikely but I guess if the delete statement that happened after this failed, and then this function gets run a second time it will raise conflicts. If that happens, do we want to treat this like an UPSERT, or should we just do nothing on the conflict. My assumption is that the inbound SMS row wouldn't get updated after creation (there is no updated_at field) so there would never be any information to update, therefore making most of this redundant and we should instead do an `on_conflict_do_nothing`?

Reply:
Oh this corresponding commit may be of slight interest to see where this code has been taken from https://github.com/alphagov/notifications-api/commit/0def0b7fd089aaa62bcc3892ddab25771e49182b

Reply:
hmmm i think i'm fine with on_conflict_do_nothing. @servingUpAces?

Reply:
It's a similar pattern to the one we use for NotificationHistory. You are correct that unlike Notifications InboundSms don't have a lifecycle to the update will not change anything (except the audit columns).  Your suggestion of using `on_conflict_do_nothing` is appropriate. We need to be able to handle the insert happening twice, in case something goes wrong during the task and it's re-run. But there should be a unit test for this.